compile_data.py

```diff
@@ -104,11 +116,26 @@ def get_evals_json(src_data_dir):
     # Summer courses are usually more intensive and thus not representative
     df = df[df["semester"] != "Summer"]
 
+    #recentsDict[courseID] = date,
+    #where date is the most recent time the course was taken
+    recentsDict = dict()
+    for ind, row in df.iterrows():
+        cid = row['course id']
+        date = row['date']
+        if cid not in recentsDict:
+            recentsDict[cid] = date
+            continue
+        else:
+            if date > recentsDict[cid]:
+                recentsDict[cid] = date
+
+    mask = df.apply(lambda row: recentsDict[row['course id']] == row['date'], axis = 1)
+    df = df[mask]
```

Comment:
The same could be accomplished with something like:

    max_dates = df['course_id', 'date'].groupby('course_id').max()
    df = df[df['date'] == df['course_id'].map(max_dates)]